src/client.rs

```diff
@@ -0,0 +1,98 @@
+use crate::error::Error;
+
+use kube::api::{Meta, PatchParams, PostParams};
+use kube::client::Client as KubeClient;
+use kube::Api;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// This `Client` can be used to access Kubernetes.
+/// It wraps an underlying [kube::client::Client] and provides some common functionality.
+#[derive(Clone)]
+pub struct Client {
+    client: KubeClient,
+    post_params: PostParams,
+    patch_params: PatchParams,
+}
+
+impl Client {
+    pub fn new(client: KubeClient) -> Self {
+        Client {
+            client,
+            post_params: PostParams::default(),
+            patch_params: PatchParams::default(),
+        }
+    }
+
+    /// Returns a [kube::client::Client]] that can be freely used.
+    /// It does not need to be cloned before first use.
+    pub fn kube_client(&self) -> KubeClient {
```

Comment:
I've looked at https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv and won't pretend to fully understand it, but could we maybe rename this to .into_inner() or .into_kube_client() ?
I _think_ that into usually consumes the wrapping object though, maybe _as_ is more appropriate here?

Reply:
Hmm I have to admit I forgot about into_* but I don't think that makes sense here. But maybe as_kube_client or to_kube_client. Preferences?

Reply:
Based on the docs I would say that "as" is correct, as this decreases the level of abstraction. "to" would indicate staying at the same level of abstraction and just changing the representation.